ZcashFoundation/zcash-grant-system
# Zcash contributions (Pt. 1 - Contribution modal)

Initial work on #15 & #16.

### What This Does

* Reworks the `contribution` model to work better with zcash
* Adds a modal that pops up when you try to contribute that gives you instructions on how to zcash
* Initial work on communication with the blockchain microservice, fetching addresses & memo for the modal

### What This Doesn't Do

* Show contributions on user profile
* Mark them as confirmed when they've been confirmed
* Update the proposal funded amount

### Steps to Test

* Wipe your database (flask shell, db.drop_all(), delete alembic table) and do a db upgrade
* Make sure your frontend, backend, and blockchain microservice are configured and running
* Make a new contribution and try to contribute to it
* Confirm the instructions shown are valid. Try sending the CLI transaction to confirm.


## Screenshots

<img width="609" alt="screen shot 2019-01-07 at 4 49 35 pm" src="https://user-images.githubusercontent.com/649992/50795446-3bdc4400-129c-11e9-9ffe-4c08cc4259a4.png">
